### PR TITLE
feat(DDI-781): Adjust dropdown menu position

### PR DESCRIPTION
**Pull request by Radha-Alberta**
- Adjusted dropdown menu to show "up" when dropdown is at bottom of the page.
- Disabled scroll when dropdown menu is open

### CODE REVIEW

File: libs/web-components/src/components/dropdown/Dropdown.svelte [hunk 1]
```diff
@@ -43,6 +44,8 @@
   let _isMenuVisible = false;
   let _highlightedIndex: number = 0;
   let _computedWidth: string;
+  let _isDropUp = false;
```
Comment (chrisolsen) [inline libs/web-components/src/components/dropdown/Dropdown.svelte:47]: This flag isn't required as it just indicates if the `_dropdownMenuPos` has been set

File: libs/web-components/src/components/dropdown/Dropdown.svelte [hunk 2]
```diff
@@ -347,7 +363,7 @@
       tabindex="0"
       class="dropdown-list"
       class:dropdown-active={_isMenuVisible}
-      style={`overflow-y: auto; max-height: ${maxheight}`}
+      style={`overflow-y: auto; max-height: ${maxheight}; --bottom: ${_isDropUp ? _dropdownMenuPos : "auto"};`}
```
Comment (chrisolsen) [inline libs/web-components/src/components/dropdown/Dropdown.svelte:366]: Can you format this a bit?
```
style={`
  overflow-y: auto; 
  max-height: ${maxheight}; 
  --bottom: ${_isDropUp ? _dropdownMenuPos : "auto"
};
```

File: libs/web-components/src/components/dropdown/Dropdown.svelte [hunk 3]
```diff
@@ -165,13 +168,25 @@
...
   function closeMenu() {
     _menuEl.removeEventListener("blur", closeMenu);
     _menuEl.removeEventListener("mouseover", onHighlight);
     setHighlightedIndexToSelected();
     _isMenuVisible = false;
+    _isDropUp = false;
+  }
+
+  function setDropdownMenuPosition() {
+    let dropdownRect = _el.getBoundingClientRect();
```
Comment (chrisolsen) [inline libs/web-components/src/components/dropdown/Dropdown.svelte:183]: Depending on your editor these `let` statements will get flagged and suggest to use `const`

**Review by chrisolsen — CHANGES_REQUESTED** [review]
I added a few comments

File: libs/web-components/src/components/dropdown/Dropdown.svelte (hunk 1 above)
Comment (Radha-Alberta) [reply libs/web-components/src/components/dropdown/Dropdown.svelte]: Removed _isDropUp flag

File: libs/web-components/src/components/dropdown/Dropdown.svelte (hunk 2 above)
Comment (Radha-Alberta) [reply libs/web-components/src/components/dropdown/Dropdown.svelte]: Done

File: libs/web-components/src/components/dropdown/Dropdown.svelte (hunk 3 above)
Comment (Radha-Alberta) [reply libs/web-components/src/components/dropdown/Dropdown.svelte]: Changed to `const`

File: libs/web-components/src/components/dropdown/Dropdown.svelte [hunk 4]
```diff
@@ -165,13 +167,24 @@
...
     setHighlightedIndexToSelected();
     _isMenuVisible = false;
+    _dropdownMenuPos = "auto";
+  }
+
+  function setDropdownMenuPosition() {
+    const dropdownRect = _el.getBoundingClientRect();
+    const menuRect = _menuEl.getBoundingClientRect();
+    if(window.innerHeight - dropdownRect.top < dropdownRect.height + menuRect.height)
+    {
```
Comment (chrisolsen) [inline libs/web-components/src/components/dropdown/Dropdown.svelte:185]: Minor nitpick: Can you use the c-style / K&R braces placement
```
if (cond) {

}
```

**Review by ArakTaiRoth — DISMISSED** [review]

File: libs/web-components/src/components/dropdown/Dropdown.svelte (hunk 4 above)
Comment (ArakTaiRoth) [reply libs/web-components/src/components/dropdown/Dropdown.svelte]: @chrisolsen This has been done

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**